### DISCUSSION TITLE
Shorted Property builder methods

### DISCUSSION THREAD

**Discussion by eeverman** [eeverman/andhow#605] [General]
Ref Issue #587 

The current validation methods are very long and wordy.  They are very clear in meaning, e.g. `mustMatchRegex`, however, it makes Property builder strings soooo looong...  I'm guessing the meanings are clear enough w/o the extra words.

I'd also like to purge deprecated methods with the 0.5.0 release, so if the old methods are going to be deprecated and renamed, perhaps it should be now in the 04.2 release.  The actual change would be relatively easy.

In general, It makes sense to follow the pattern of existing methods.  For String that is easy.  Here are the current validation methods in [StrProp](https://github.com/eeverman/andhow/blob/main/andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java) and proposed new names:
| Current Name | Proposed New Name | Notes |
| ----------------- | --------------------- | ------------------------- |
| mustMatchRegex | matches | Same as `String.matches(regex)` |
| mustStartWith      | startsWith | Same as `String.startsWith(prefix)` |
| mustStartWith      | startsWith | Same as `String.startsWith(prefix)` |
| mustStartWithIgnoreCase | startsWithIgnoreCase | extension of startsWith |
| mustEndWith       | endsWith | Same as `String.endsWith(suffix)` (sufix is misspelled on this method) |
| mustEndWithIgnoreCase | endsWithIgnoreCase | extension of startsWith |
| mustEqual(String...) | oneOf(String...) | 'equals' cannot be used - There is no exact corollary. |
| mustBeNonNull | notNull() | No comp in the String class, but Junit uses `assertNotNull`

Missing from that list is `oneOfIgnoreCase` - This method is missing and should be another ticket.

Things are less clear for numeric Properties like IntProp, BigDecProp, etc.  There is no obvious example to follow that I can think of.  Here are some proposed method names a numeric Property, but I've no idea what the best and most clear names would be:
| Current Name | Proposed New Name | Notes |
| ----------------- | --------------------- | ------------------------- |
| mustBeGreaterThan | greaterThan |  |
| mustBeGreaterThanOrEqualTo      | greaterThanOrEqual |  |
| mustBeLessThan      | lessThan |  |
| mustBeLessThanOrEqualTo | lessThanOrEqual |  |

**Comment by eeverman** [comment]
@alex-kar Can you comment on this?  All the xxxProp classes are [here](https://github.com/eeverman/andhow/tree/homepage/andhow-core/src/main/java/org/yarnandtail/andhow/property) for reference.

**Reply by alex-kar** [reply]
I agree, meanings are clear enough even without the extra words.
What about names from Mockito matchers for numeric Properties?
I don't really like it, just want to offer as an alternative.
| Current Name  | Proposed New Name |
| ------------- | ------------- |
| mustBeGreaterThan  | gt  |
| mustBeGreaterThanOrEqualTo  | geq  |
| mustBeLessThan  | lt  |
| mustBeLessThanOrEqualTo  | leq  |

**Reply by eeverman** [reply]
@alex-kar  Good thought - But I agree its perhaps a bit too terse.  I wonder if those short method names could be added for people who want to use a more compact syntax?  It might be a good future addition.

Mockito got me thinking of Hamcrest which I believe is in common use w/ JUnit.  The [Matchers](http://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/Matchers.html) class is very close to the AndHow method names with the 'must' removed:

| Current Name | Proposed New Name | Notes |
| -- | -- | -- |
| mustBeGreaterThan | greaterThan |  |
| mustBeGreaterThanOrEqualTo | greaterThanOrEqual**To** |  Hamcrest adds 'To' |
| mustBeLessThan | lessThan |   |
| mustBeLessThanOrEqualTo | lessThanOrEqual**To** |  Hamcrest adds 'To' |

Hamcrest also has String matching that is close but slightly different.  Here are key ones from their string matchers:

| Current Name | Proposed New Name | Notes |
| -- | -- | -- |
| mustEndWithIgnoreCase | endsWithIgnor**ing**Case | Hamcrest has the 'ing'  |
| mustStartWithIgnoreCase | startsWithIgnor**ing**Case | Hamcrest has the 'ing' |
| mustEqual(String...) | oneOf(String...) | Same as above, but Hamcrest has it as well, so an added vote  |

So, I propose following the Hamcrest names for numerics and the String validation that have no corollary in the String class.  What do you think?  Are you interested in implementing it?

**Reply by alex-kar** [reply]
@eeverman Yes, I would like to take it. This task #587 just to rename those methods, but you mentioned missing `oneOfIgnoreCase`. Can you assign it to me as well if you're going to open new ticket? Thanks.
